aptos-move/aptos-vm/src/block_executor/mod.rs

```diff
@@ -471,10 +432,34 @@ impl BlockAptosVM {
 
         BLOCK_EXECUTOR_CONCURRENCY.set(config.local.concurrency_level as i64);
 
-        let environment = get_environment_with_delayed_field_optimization_enabled(
-            state_view,
-            global_module_cache.as_ref(),
-        )?;
+        let environment =
+            AptosEnvironment::new_with_delayed_field_optimization_enabled(&state_view);
```

Comment:
Yes, if the new configuration is different. I am keeping it outside for feature gating (need to get the flag for V2 loader), but it should go inside `check_ready_...` - we fetch environment from state view and compare with whatever we cached. If it is the same, use the cached one. Otherwise - update and flush.